src/nu/validator/htmlparser/impl/Tokenizer.java

```diff
@@ -3181,6 +3181,12 @@ private void ensureBufferSpace(int inputLength) throws SAXException {
                             case '>':
                                 cstart = pos + 1;
                                 state = transition(state, Tokenizer.DATA, reconsume, pos);
+                                // Since a CDATA section starts with a less-than sign, it
+                                // is participates in the suspension-after-current-token
+                                // behavior. (The suspension can be requested when the
+                                // less-than sign has been seen but we don't yet know the
+                                // resulting token type.) Therefore, we need to deal with
+                                // a potential request here.
                                 suspendIfRequestedAfterCurrentNonTextToken();
```

Comment:
```suggestion
                                // Since a CDATA section starts with a less-than sign, it
                                // participates in the suspension-after-current-token
                                // behavior. (The suspension can be requested when the
                                // less-than sign has been seen but we don't yet know the
                                // resulting token type.) Therefore, we need to deal with
                                // a potential request here.
                                suspendIfRequestedAfterCurrentNonTextToken();
```